feature_engine/encoding/count_frequency.py

```diff
@@ -143,16 +148,44 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
 
         y: pandas Series, default = None
             y is not needed in this encoder. You can pass y or None.
+
+        threshold : list of floats
+            Thresholds for each column needed for binning rare categories
         """
         X = check_X(X)
         self._check_or_select_variables(X)
         self._get_feature_names_in(X)
 
         self.encoder_dict_ = {}
 
+        if not isinstance(threshold, list):
+            threshold = [None] * len(self.variables_)
+
         # learn encoding maps
-        for var in self.variables_:
-            if self.encoding_method == "count":
+        for var, threshold in zip(self.variables_, threshold):
```

Comment:
I would let the user pass 1 threshold, at this would be applied to all variables. If they want different thresholds for different variable groups, then they can use 2 different transformers.

Otherwise, the class becomes too complicated.